Arm/BitVec.lean

```diff
@@ -270,14 +270,15 @@ protected def rand (n : Nat) (lo := 0) (hi := 2^n - 1) : IO (BitVec n) := do
 def unsigned_compare (a b : BitVec n) : Ordering :=
   if BitVec.ult a b then .lt else if a = b then .eq else .gt
 
+@[bitvec_rules]
 abbrev ror (x : BitVec n) (r : Nat) : BitVec n :=
   rotateRight x r
 
 /-- Return the `i`-th least significant bit (or `0` if `i >= w`) of
     the `n`-bit bitvector `x`. -/
 @[bitvec_rules]
 abbrev lsb (x : BitVec n) (i : Nat) : BitVec 1 :=
-  BitVec.ofBool (getLsb x i)
+  (BitVec.extractLsb i i x).cast (by omega)
```

Comment:
If you use `extractLsb'` (which is phrased in terms of `lo` and length, rather than `hi-lo`) you don't need the cast

Reply:
Indeed. Unfortunately, `extractLsb'` doesn't seem to be supported by LeanSAT. Here's an illustration:

```
theorem extractLsb_and_Lsb' (x : BitVec 32) :
  BitVec.extractLsb 31 31 x = BitVec.extractLsb' 31 1 x := by
  /-
  -- bv_decide
  Spurious counterexample:
  The prover found a potential counter example, consider the following assignment:
  x = 0xffffffff#32
  BitVec.extractLsb' 31 1 x = 0x0#1
  -/
  simp only [Nat.reduceSub, Nat.reduceAdd, BitVec.extractLsb]
```

Reply:
And upon leaving out `(x : BitVec 32)` above, we get a different error message from LeanSAT:
`None of the hypotheses are in the supported BitVec fragment....`

Reply:
Reported to Henrik on Zulip.